chain/client/tests/ganache.rs

```diff
@@ -0,0 +1,173 @@
+use std::iter::FromIterator;
+use std::path::Path;
+use std::str::FromStr;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::{Arc, RwLock};
+
+use actix::System;
+use futures::{future, FutureExt};
+use num_rational::Rational;
+
+use near_actix_test_utils::run_actix_until_stop;
+use near_chain::chain::NUM_EPOCHS_TO_KEEP_STORE_DATA;
+use near_chain::types::LatestKnown;
+use near_chain::validate::validate_chunk_with_chunk_extra;
+use near_chain::{
+    Block, ChainGenesis, ChainStore, ChainStoreAccess, ErrorKind, Provenance, RuntimeAdapter,
+};
+use near_chain_configs::{ClientConfig, Genesis};
+use near_chunks::{ChunkStatus, ShardsManager};
+use near_client::test_utils::{create_chunk_on_height, setup_mock_all_validators};
+use near_client::test_utils::{setup_client, setup_mock, TestEnv};
+use near_client::{Client, GetBlock, GetBlockWithMerkleTree};
+use near_crypto::{InMemorySigner, KeyType, PublicKey, Signature, Signer};
+use near_logger_utils::init_test_logger;
+#[cfg(feature = "metric_recorder")]
+use near_network::recorder::MetricRecorder;
+use near_network::routing::EdgeInfo;
+use near_network::test_utils::{wait_or_panic, MockNetworkAdapter};
+use near_network::types::{NetworkInfo, PeerChainInfoV2, ReasonForBan};
+use near_network::{
+    FullPeerInfo, NetworkClientMessages, NetworkClientResponses, NetworkRequests, NetworkResponses,
+    PeerInfo,
+};
+use near_primitives::block::{Approval, ApprovalInner};
+use near_primitives::block_header::BlockHeader;
+use near_primitives::errors::InvalidTxError;
+use near_primitives::hash::{hash, CryptoHash};
+use near_primitives::merkle::verify_hash;
+#[cfg(not(feature = "protocol_feature_block_header_v3"))]
+use near_primitives::sharding::ShardChunkHeaderV2;
+use near_primitives::sharding::{EncodedShardChunk, ReedSolomonWrapper, ShardChunkHeader};
+#[cfg(feature = "protocol_feature_block_header_v3")]
+use near_primitives::sharding::{ShardChunkHeaderInner, ShardChunkHeaderV3};
+
+use near_primitives::receipt::DelayedReceiptIndices;
+use near_primitives::serialize::from_base64;
+use near_primitives::state_record::StateRecord;
+use near_primitives::syncing::{get_num_state_parts, ShardStateSyncResponseHeader};
+use near_primitives::transaction::{
+    Action, DeployContractAction, ExecutionStatus, FunctionCallAction, SignedTransaction,
+    Transaction,
+};
+use near_primitives::trie_key::TrieKey;
+use near_primitives::types::validator_stake::ValidatorStake;
+use near_primitives::types::{AccountId, BlockHeight, EpochId, NumBlocks, StoreKey};
+use near_primitives::utils::to_timestamp;
+use near_primitives::validator_signer::{InMemoryValidatorSigner, ValidatorSigner};
+use near_primitives::version::PROTOCOL_VERSION;
+use near_primitives::views::{
+    BlockHeaderView, FinalExecutionStatus, QueryRequest, QueryResponseKind,
+};
+use near_store::get;
+use near_store::test_utils::create_test_store;
+use neard::config::{GenesisExt, TESTING_INIT_BALANCE, TESTING_INIT_STAKE};
+use neard::NEAR_BASE;
```

Comment:
Unrelated to the PR, but it seems that there's something wrong with our testing infra if adding a simple smoke test for new functionality requires more imports than fit on my screen. 

I think this smell points to the problem that we use *code* to setup the test. The better solution is usually to do setup by specifying *data* which is required, and having a single generic driver that interprets the data and sets up the in-memory objects. `NodeCluster` refactor was a great step in that direction:

https://github.com/near/nearcore/blob/0687126597dbf9c61af33ee270086d60dd645911/neard/tests/rpc_nodes.rs#L51-L57

Reply:
I usually copy-paste the code without imports and then <kbd>Alt+Enter</kbd> missing ones.

Reply:
Removed. Alt+Enter is quite helpful